src/verilog.y

```diff
@@ -1284,9 +1284,31 @@ portsStarE<nodep>:		// IEEE: .* + list_of_ports + list_of_port_declarations + em
 	|	'(' {VARRESET_LIST(PORT);} list_of_ports ')'	{ $$ = $3; VARRESET_NONLIST(UNKNOWN); }
 	;
 
+list_of_portsEmpty<nodep>:		// IEEE: list_of_ports + list_of_port_declarations
+		portAndTagEmpty			{ $$ = $1; }
+	|	list_of_portsEmpty ',' portAndTagEmpty		{ $$ = $1->addNextNull($3); }
+	;
+
 list_of_ports<nodep>:		// IEEE: list_of_ports + list_of_port_declarations
-		portAndTag				{ $$ = $1; }
-	|	list_of_ports ',' portAndTag		{ $$ = $1->addNextNull($3); }
+		portAndTag			{ $$ = $1; }
+	|	list_of_portsEmpty ',' portAndTagEmpty		{ $$ = $1->addNextNull($3); }
+	;
+
+portAndTagEmpty<nodep>:
+		/* empty */
+			{ int p = PINNUMINC();
+			  string pn = "__pinNumber"+cvtToStr(p);
+			  $$ = new AstPort(CRELINE(), p, pn);
+			  AstVar* varp = new AstVar(CRELINE(), AstVarType::PORT, pn, VFlagChildDType(),
+			                            new AstBasicDType(CRELINE(), LOGIC_IMPLICIT));
```

Comment:
```suggestion
			   const string name = "__pinNumber" + cvtToStr(p);
			  $$ = new AstPort{CRELINE(), p, name};
			  AstVar* varp = new AstVar{CRELINE(), AstVarType::PORT, name, VFlagChildDType{},
			                            new AstBasicDType{CRELINE(), LOGIC_IMPLICIT}};
```

Reply:
sure, I'll fix the const.
As for the C++ initializer list constructor, may I ask why?
Found [this](https://stackoverflow.com/questions/47366453/direct-initialization-vs-direct-list-initialization-c) and [this](https://en.cppreference.com/w/cpp/language/list_initialization) (direct list initialization 3).
Given that you have no constructor with std::initializer_list<>, why does it matter?

Reply:
One more question:
I'm using `name` for both the AstPort and the AstVar objects. Is there any section in the code where these objects are freed, other than the normal program termination?
If this was some other program with more dynamic behavior, I would be worried about freeing `name` twice when these objects get deallocated.

Reply:
As to string, C++ handles all memory for strings, so no need to deallocate.

As to {}, we're trying to use it with new code but didn't retrofit old code, see here https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives